x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -213,6 +210,13 @@ ResolvedIndices resolveIndicesAndAliases(IndicesRequest indicesRequest, MetaData
             } else {
                 resolvedIndicesBuilder.addLocal(aliasesRequest.aliases());
             }
+            // if no aliases are authorized, then fill in an expression that
+            // MetaData#findAliases evaluates to the empty alias list. You cannot put
+            // "nothing" (the empty list) explicitly because this is resolved by es core to
+            // _all
+            if (aliasesRequest.aliases().length == 0) {
+                aliasesRequest.replaceAliases(NO_INDICES_OR_ALIASES_ARRAY);
+            }
```

Comment:
This is the gist.

Reply:
@jasontedor @tvernum  I believe this is the root issue!
It should be:
```
if (indicesRequest instanceof GetAliasesRequest && aliasesRequest.aliases().length == 0) {
    aliasesRequest.replaceAliases(NO_INDICES_OR_ALIASES_ARRAY);
}
```
For the remove index alias action type the `alias` parameter is not used and replacing it triggers the bug.